torch/utils/_python_dispatch.py

```diff
@@ -0,0 +1,33 @@
+import torch
+import contextlib
+from typing import Iterator
+
+# Context manager that causes all pytorch operators to dispatch to the passed-in
+# type's __torch_dispatch__ function.
+# operation that accepts no tensors but returns a tensor.
+#
+# enable_python_mode is affected by torch._C._DisableTorchDispatch.
+#
+# NB: Calling an operator inside __torch_dispatch__ does go through
+# __torch_dispatch__ again. Please use _DisableTorchDispatch inside
+# __torch_dispatch__ to prevent infinite recursion.
+#
+# TODO: Limitations and things about enable_python_mode we should fix before exposing it:
+# - it currently cannot be nested. This should be simple to implement; we need a
+#   stack of TorchDispatchTypeObjects and the next bullet point.
+# - We need a better user-facing api for torch._C._DisableTorchDispatch that
+#   is able to selectively disable __torch_dispatch__ of a particular class.
+# - It doesn't work with the tensor constructors (torch.tensor, torch.Tensor)
+@contextlib.contextmanager
+def enable_python_mode(cls) -> Iterator[None]:
```

Comment:
I am wondering if we want a more descriptive name for this? In particular mentioning that it changes the default Tensor subclass generated by ops?

Reply:
This isn't being exposed yet so we don't need a better name for it in this PR. I think we do want a more descriptive name (because "mode" only makes sense to us); do you have ideas?

> it changes the default Tensor subclass generated by ops

I wouldn't say it changes the default Tensor subclass, it calls into the `__torch_dispatch__` of the Tensor subclass and then that does whatever it wants. The `__torch_dispatch__` doesn't have to return an instance of the Tensor subclass if it doesn't want to

Reply:
"dispatch_override"? "omni_dispatch_override"?

Reply:
default dispatch maybe?

> This isn't being exposed yet so we don't need a better name for it in this PR.

While I agree with that, it is started to get confusing even for me to have multiple keys/backend systems all called the same :D

Reply:
I added this to the list of things to follow up on (in the TODO for `torch/utils/_python_dispatch.py`). I want to get this PR in first because it's difficult to land and then we can figure out the naming in the next PR. How does that sound?

Reply:
Sure.